src/ppom-storage.ts

```diff
@@ -0,0 +1,204 @@
+import { calculateSHA256 } from './crypto-utils';
+
+type FileInfo = {
+  name: string;
+  chainId: string;
+  version: string;
+  checksum: string;
+};
+
+export type PPOMStorageMetadata = FileInfo[];
+
+export type StorageKey = {
+  name: string;
+  chainId: string;
+};
+
+export type StorageBackend = {
+  read(key: StorageKey): Promise<any>;
+  write(key: StorageKey, data: any): Promise<void>;
```

Comment:
Since `#validateChecksum` takes an ArrayBuffer, should this be the following?

``` suggestion
  read(key: StorageKey): Promise<ArrayBuffer>;
  write(key: StorageKey, data: ArrayBuffer): Promise<void>;
```